drivers/net/ethernet/intel/ice/ice_flow.c

```diff
@@ -1559,26 +1697,25 @@ ice_flow_disassoc_prof(struct ice_hw *hw, enum ice_block blk,
 #define FLAG_GTPU_DW	FLAG_GTP_EH_PDU
 
 /**
- * ice_flow_set_parser_prof - Set flow profile based on the parsed profile info
+ * ice_flow_set_hw_prof - Set HW flow profile based on the parsed profile info
  * @hw: pointer to the HW struct
- * @dest_vsi: dest VSI
- * @fdir_vsi: fdir programming VSI
+ * @dest_vsi_handle: dest VSI handle
+ * @fdir_vsi_handle: fdir programming VSI handle
  * @prof: stores parsed profile info from raw flow
- * @blk: classification blk
- *
- * Return: 0 on success or negative errno on failure.
+ * @blk: classification stage
  */
 int
-ice_flow_set_parser_prof(struct ice_hw *hw, u16 dest_vsi, u16 fdir_vsi,
-			 struct ice_parser_profile *prof, enum ice_block blk)
+ice_flow_set_hw_prof(struct ice_hw *hw, u16 dest_vsi_handle,
+		     u16 fdir_vsi_handle, struct ice_parser_profile *prof,
+		     enum ice_block blk)
 {
-	u64 id = find_first_bit(prof->ptypes, ICE_FLOW_PTYPE_MAX);
-	struct ice_flow_prof_params *params __free(kfree);
+	int id = find_first_bit(prof->ptypes, ICE_FLOW_PTYPE_MAX);
+	struct ice_flow_prof_params *params;
```

Comment:
double check if all error patch properly release @params (as you removed __free()).
or just undo